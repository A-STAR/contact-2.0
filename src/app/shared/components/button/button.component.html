<button [ngClass]="buttonClass" [disabled]="disabled" [title]="displayTitle | translate">
  <i *ngIf="iconClass" [ngClass]="iconClass" aria-hidden="true"></i>
  <span *ngIf="displayLabel">{{ displayLabel | translate }}</span>
</button>
