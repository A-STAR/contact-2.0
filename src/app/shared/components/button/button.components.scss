:host {
  button {
    // &:not(.btn-icon) {
    //   background-color: #58526e;
    //   color: #fff;
    // }

    &.btn-arrow {
      position: relative;
      height: 34px;
      background-color: #58526e;
      color: #fff;
      font-size: 15px !important;
      border-top-right-radius: 0 !important;
      border-bottom-right-radius: 0 !important;
      line-height: 34px;

      &::before,
      &::after {
        position: absolute;
        top: 0;
        content: '';
        width: 0;
        height: 0;
        border-top: 16px solid transparent;
        border-bottom: 16px solid transparent;
        border-left: 11px solid transparent;
      }

      &::before {
        right: -22px;
        border-left-color: #44a1f8;
      }

      &::after {
        right: -11px;
        border-color: #44a1f8;
        border-left-color: #58526e;
      }

    }

  }

}
