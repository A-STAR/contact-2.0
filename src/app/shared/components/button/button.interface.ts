export type IButtonStyle =
  'none' |
  'default' |
  'primary' |
  'secondary' |
  'warning' |
  'danger';

export type IButtonType =
  'add' |
  'addUser' |
  'back' |
  'block' |
  'cancel' |
  'change' |
  'changeStatus' |
  'clear' |
  'close' |
  'copy' |
  'delete' |
  'download' |
  'edit' |
  'exportToExcel' |
  'move' |
  'next' |
  'ok' |
  'refresh' |
  'registerContact' |
  'save' |
  'search' |
  'select' |
  'sms' |
  'unblock' |
  'undo' |
  'upload' |
  'visit'|
  'start' |
  'stop';

export type IButtonStylesConfig = {
  [K in IButtonStyle]: string;
};

export type IButtonTypesConfig = {
  [K in IButtonType]: {
    icon: string;
    label: string;
  };
};
