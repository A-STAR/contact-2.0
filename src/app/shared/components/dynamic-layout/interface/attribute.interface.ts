import { IDynamicLayoutGenericItem, DynamicLayoutItemType } from './item.interface';

export interface IAttributeContext {
  debtId: number;
  personId: number;
  personRole: number;
  userId: number;
}

export enum AttributeType {
  FORMULA    = 'formula',
  FORMULA_ID = 'formulaId',
}

export interface IAttribute {
  type: AttributeType;
  value: string;
}

export interface IAttributeConfig {
  [key: string]: IAttribute[];
}

export interface IAttributePayload {
  attributes: IAttributeConfig;
  context: Partial<IAttributeContext>;
}

export interface IDynamicLayoutAttribute extends IDynamicLayoutGenericItem {
  attributeType: AttributeType;
  type: DynamicLayoutItemType.ATTRIBUTE;
  value: string;
  // Optional:
  dictCode?: number;
  lookupKey?: number;
}
