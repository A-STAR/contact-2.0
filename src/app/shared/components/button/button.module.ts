import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TranslateModule } from '@ngx-translate/core';

import { ButtonService } from './button.service';

import { ButtonComponent } from './button.component';

@NgModule({
  imports: [
    CommonModule,
    TranslateModule,
  ],
  exports: [
    ButtonComponent,
  ],
  declarations: [
    ButtonComponent,
  ],
  providers: [
    ButtonService,
  ]
})
export class ButtonModule { }
