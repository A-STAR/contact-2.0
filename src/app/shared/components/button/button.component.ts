import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { IButtonStyle } from './button.interface';

import { ButtonService } from './button.service';
import { IconType } from '@app/shared/components/icons/icons.interface';

@Component({
  selector: 'app-button',
  templateUrl: 'button.component.html',
  styleUrls: ['./button.components.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ButtonComponent {
  @Input() color: IButtonStyle = 'default';
  @Input() disabled: boolean;
  @Input() icon: string;
  @Input() label: string | false;
  @Input() title: string;
  @Input() type: IconType;
  // TODO(i.lobanov): remove it when btn service is refactored
  @Input() btnClass: string;
  @Input() withBtnClass = true;

  constructor(private buttonService: ButtonService) {}

  get buttonClass(): string {
    return (this.btnClass ? this.btnClass + ' ' : '')
      // + (this.iconClass ? 'btn-icon ' : '')
      + this.buttonService.getClass(this.color, this.withBtnClass);
  }

  get iconClass(): string {
    return this.icon || this.buttonService.getIcon(this.type);
  }

  get displayLabel(): string {
    return this.label === false ? null : (this.label || this.buttonService.getLabel(this.type));
  }

  get displayTitle(): string {
    return this.title || this.label || this.buttonService.getLabel(this.type);
  }

  onClick(event: MouseEvent): void {
    if (this.disabled) {
      event.stopPropagation();
    }
  }
}
