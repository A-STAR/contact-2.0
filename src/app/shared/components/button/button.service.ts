import { Injectable } from '@angular/core';

import { IButtonStyle, IButtonStylesConfig, IButtonType, IButtonTypesConfig } from './button.interface';

@Injectable()
export class ButtonService {
  private defaultIcons: IButtonTypesConfig = {
    add:              { icon: 'fa fa-plus',             label: 'default.buttons.add'              },
    addUser:          { icon: 'fa fa-user-plus',        label: 'default.buttons.addUser'          },
    back:             { icon: 'fa fa-arrow-left',       label: 'default.buttons.back'             },
    block:            { icon: 'fa fa-unlock-alt',       label: 'default.buttons.block'            },
    cancel:           { icon: 'fa fa-ban',              label: 'default.buttons.cancel'           },
    change:           { icon: '',                       label: 'default.buttons.change'           },
    changeStatus:     { icon: 'fa fa-random',           label: 'default.buttons.changeStatus'     },
    clear:            { icon: 'fa fa-times',            label: 'default.buttons.clear'            },
    close:            { icon: 'fa fa-ban',              label: 'default.buttons.close'            },
    copy:             { icon: 'fa fa-clone',            label: 'default.buttons.copy'             },
    delete:           { icon: 'fa fa-trash',            label: 'default.buttons.delete'           },
    download:         { icon: 'fa fa-cloud-download',   label: 'default.buttons.download'         },
    email:            { icon: 'fa fa-envelope',         label: 'default.buttons.email'            },
    edit:             { icon: 'fa fa-pencil',           label: 'default.buttons.edit'             },
    exportToExcel:    { icon: 'fa fa-file-excel-o',     label: 'default.buttons.exportToExcel'    },
    move:             { icon: 'fa fa-share',            label: 'default.buttons.move'             },
    next:             { icon: 'fa fa-arrow-right',      label: 'default.buttons.next'             },
    ok:               { icon: 'fa fa-check',            label: 'default.buttons.ok'               },
    refresh:          { icon: 'fa fa-refresh',          label: 'default.buttons.refresh'          },
    registerContact:  { icon: 'fa fa-tty',              label: 'default.buttons.registerContact'  },
    save:             { icon: 'fa fa-save',             label: 'default.buttons.save'             },
    search:           { icon: 'fa fa-search',           label: 'default.buttons.search'           },
    select:           { icon: '',                       label: 'default.buttons.select'           },
    start:            { icon: 'fa fa-play',             label: 'default.buttons.start'            },
    stop:             { icon: 'fa fa-stop',             label: 'default.buttons.stop'             },
    sms:              { icon: 'fa fa-envelope',         label: 'default.buttons.sms'              },
    unblock:          { icon: 'fa fa-unlock',           label: 'default.buttons.unblock'          },
    undo:             { icon: 'fa fa-undo',             label: 'default.buttons.undo'             },
    upload:           { icon: 'fa fa-cloud-upload',     label: 'default.buttons.upload'           },
    version:          { icon: 'fa fa-code-fork',        label: 'default.buttons.version'          },
    visit:            { icon: 'fa fa-bus',              label: 'default.buttons.visit'            },
  };

  private styles: IButtonStylesConfig = {
    none: 'btn btn-link',
    default: 'btn btn-default',
    primary: 'btn btn-success',
    secondary: 'btn btn-primary',
    warning: 'btn btn-warning',
    danger: 'btn btn-danger',
  };

  getIcon(type: IButtonType): string {
    return this.defaultIcons[type] ? this.defaultIcons[type].icon : null;
  }

  getLabel(type: IButtonType): string {
    return this.defaultIcons[type] ? this.defaultIcons[type].label : null;
  }

  getClass(style: IButtonStyle): string {
    return this.styles[style];
  }
}
