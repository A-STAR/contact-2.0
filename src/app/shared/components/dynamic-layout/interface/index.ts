import { Observable } from 'rxjs/Observable';

import { IDynamicLayoutControl } from './control.interface';
import { IDynamicLayoutGroup } from './group.interface';
import { IDynamicLayoutTemplate } from './template.interface';

export * from './context.interface';
export * from './control.interface';
export * from './attribute.interface';
export * from './group.interface';
export * from './item.interface';
export * from './template.interface';

export type IDynamicLayoutItem = IDynamicLayoutGroup | IDynamicLayoutControl | IDynamicLayoutTemplate;

export interface IDynamicLayoutConfig {
  items: IDynamicLayoutItem[];
}

export interface IDynamicLayoutItemProperties {
  item: IDynamicLayoutItem;
  streams: {
    display: Observable<boolean>;
  };
}
