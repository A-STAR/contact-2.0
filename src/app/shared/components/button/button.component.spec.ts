import { async as Async, ComponentFixture, TestBed } from '@angular/core/testing';
import { TranslateLoader, TranslateModule } from '@ngx-translate/core';
import { Observable } from 'rxjs/Observable';
import { of } from 'rxjs/observable/of';

import { IButtonStyle, IButtonType } from './button.interface';

import { ButtonService } from './button.service';

import { ButtonComponent } from './button.component';

class ButtonServiceMock {
  getIcon(type: IButtonType): string {
    return 'default-icon-class';
  }

  getLabel(type: IButtonType): string {
    return 'default-label';
  }

  getClass(style: IButtonStyle): string {
    return 'default-button-class';
  }
}

class TranslateLoaderMock {
  getTranslation(language: string): Observable<any> {
    return of([]);
  }
}

describe('ButtonComponent', () => {
  let fixture: ComponentFixture<ButtonComponent>;

  beforeEach(Async(() => {
    TestBed
      .configureTestingModule({
        declarations: [
          ButtonComponent,
        ],
        imports: [
          TranslateModule.forRoot({
            loader: {
              provide: TranslateLoader,
              useClass: TranslateLoaderMock,
            },
          }),
        ],
        providers: [
          {
            provide: ButtonService,
            useClass: ButtonServiceMock,
          },
        ],
      })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ButtonComponent);
  });

  it('should render button', () => {
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should render button with custom icon class', () => {
    fixture.componentInstance.icon = 'custom-icon-class';
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should render button with no label', () => {
    fixture.componentInstance.label = false;
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should render button with custom label', () => {
    fixture.componentInstance.label = 'Custom Label';
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
