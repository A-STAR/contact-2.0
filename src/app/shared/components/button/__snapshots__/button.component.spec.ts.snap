// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent should render button 1`] = `
<div
  id="root0"
  ng-version="5.2.1"
>
  <div
    class="inline"
  >
    
  
    <button
      class="default-button-class"
      ng-reflect-ng-class="default-button-class"
      title="default-label"
    >
      
    
      
      <i
        aria-hidden="true"
        class="default-icon-class"
        ng-reflect-ng-class="default-icon-class"
      />
      
    
      
      <span>
        default-label
      </span>
      
  
    </button>
    

  </div>
  

</div>
`;

exports[`ButtonComponent should render button with custom icon class 1`] = `
<div
  id="root1"
  ng-version="5.2.1"
>
  <div
    class="inline"
  >
    
  
    <button
      class="default-button-class"
      ng-reflect-ng-class="default-button-class"
      title="default-label"
    >
      
    
      
      <i
        aria-hidden="true"
        class="custom-icon-class"
        ng-reflect-ng-class="custom-icon-class"
      />
      
    
      
      <span>
        default-label
      </span>
      
  
    </button>
    

  </div>
  

</div>
`;

exports[`ButtonComponent should render button with custom label 1`] = `
<div
  id="root3"
  ng-version="5.2.1"
>
  <div
    class="inline"
  >
    
  
    <button
      class="default-button-class"
      ng-reflect-ng-class="default-button-class"
      title="Custom Label"
    >
      
    
      
      <i
        aria-hidden="true"
        class="default-icon-class"
        ng-reflect-ng-class="default-icon-class"
      />
      
    
      
      <span>
        Custom Label
      </span>
      
  
    </button>
    

  </div>
  

</div>
`;

exports[`ButtonComponent should render button with no label 1`] = `
<div
  id="root2"
  ng-version="5.2.1"
>
  <div
    class="inline"
  >
    
  
    <button
      class="default-button-class"
      ng-reflect-ng-class="default-button-class"
      title="default-label"
    >
      
    
      
      <i
        aria-hidden="true"
        class="default-icon-class"
        ng-reflect-ng-class="default-icon-class"
      />
      
    
      
      
  
    </button>
    

  </div>
  

</div>
`;
